crowbar_framework/app/models/dns_service.rb

```diff
@@ -123,6 +123,7 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
       node.set[:dns] = {} if node[:dns].nil?
       unless node[:dns][:master]
         node.set[:dns][:master] = true
+        node.set[:dns][:master_ip] = node.get_network_by_type("admin")["address"]
```

Comment:
this is for non-master nodes to look up the ip of the master ip. should we set it to the same node actually? why not simply fix the designate barclamp to just not look for master_ip if it is master?